### ISSUE TITLE
Automatically translate Incident titles

### ISSUE THREAD

**Issue by cesarvarela** [responsible-ai-collaborative/aiid#976]
Right now only reports' text and description are translated automatically. 

Incident titles' (and maybe other fields?) should be added to the translate process.

This is a left over from #695 .

### PRs
- https://github.com/responsible-ai-collaborative/aiid/pull/3348
- https://github.com/responsible-ai-collaborative/aiid/pull/3471

**Comment by smcgregor** [comment]
Rating this one as low priority primarily due to the amount of time this will take to implement.

**Comment by kepae** [comment]
This is now in sight. Now that we have updated our GraphQL API and are free from Atlas-imposed structure on documents, we may be able to update (/refactor) how translations are stored to handle incidents more easily, as well as adding new arbitrary languages. We no longer require materializing a named query endpoint for each language and can instead provide translation language as a query parameter.

@pdcp1 I would propose we (1) identify how we can generalize the translation pipeline and take advantage of the above (2) apply the new generalized querying (and possibly document storage) to reports and then finally (3) apply the same generic process for incidents titles and descriptions.

_My kingdom, my kingdom for i18n SEO!_

**Comment by pdcp1** [comment]
@kepae Is it safe to assume that the incident titles and descriptions are always written in English? I'm asking this because, with reports, we store the original language in the database and only translate them to English if they are not originally in English. 

However, we do not have a 'language' field for incidents. Since our editors are responsible for writing the titles and descriptions of those incidents, we could consider English as the default language. This approach would help us avoid unnecessary translations to English.

**Comment by kepae** [comment]
For now, let's assume that all incident object strings that AIID creates as part of editing (e.g. titles, descriptions, notes) are in English, and that we do not need to request English language translations for these documents.

In the future, if we need, we can add language tags to these fields and then remove that assumption -- but I don't think this should affect too much how we generate incident document translations. Let me know if I'm missing something!

**Comment by pdcp1** [comment]
That's correct. In the future, if we want to remove that assumption, we will need to add a `language` field to the `incidents` collection and handle them the same way we currently handle Reports translations.
